app/controllers/nonsipp/common/CompanyRecipientCrnController.scala

```diff
@@ -78,9 +80,12 @@ class CompanyRecipientCrnController @Inject()(
               .successful(BadRequest(view(formWithErrors, viewModel(srn, index, mode, subject, request.userAnswers)))),
           value =>
             for {
-              updatedAnswers <- Future
-                .fromTry(request.userAnswers.set(CompanyRecipientCrnPage(srn, index, subject), ConditionalYesNo(value)))
-              _ <- saveService.save(updatedAnswers)
+              updatedAnswers <- request.userAnswers
+                .set(CompanyRecipientCrnPage(srn, index, subject), ConditionalYesNo(value))
+                .mapK
+              nextPage = navigator.nextPage(CompanyRecipientCrnPage(srn, index, subject), mode, updatedAnswers)
+              updatedProgressAnswers <- saveProgress(srn, index, updatedAnswers, nextPage)
+              _ <- saveService.save(updatedProgressAnswers)
```

Comment:
Controllers in common package should take IdentitySubject into consideration. Otherwise this saves progress into `loansProgress` even if you do a section other than loans that uses this controller. E.g. Land or property seller:
![image](https://github.com/user-attachments/assets/483305ca-0f2f-4626-aa6a-094753ce98b9)

Reply:
Maybe it would need a `saveProgress` method in `common` package where you pass in `identitySubject` and route to `loansmadeoroutstanding.saveProgress` only if identity subject is `loanRecipient` (something like this selection [here](https://github.com/hmrc/pension-scheme-return-frontend/blob/c0f499890b9307d64eda20e5b530bd072446c761/app/pages/nonsipp/common/OtherRecipientDetailsPage.scala#L30) ).